### PR TITLE
sar.json Initial Buildout

### PR DESCRIPTION
**Pull request by karla-vm**
### Description

Getting a head start on the SAR form pages.

### Related ticket(s)

MDCT-2963

---
### How to test

- Navigate to `/sar`
- Generate a new SAR report
- Go through the pages, make sure everything looks good

### Important updates

N/A

---
### Author checklist

- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

File: services/app-api/forms/sar.json [hunk 1]
```diff
@@ -14,10 +14,143 @@
...
+            "type": "text",
+            "validation": "text",
+            "props": {
+              "label": "AOR title/agency"
+            }
+          },
+          {
+            "id": "generalInformation_aorEmail",
+            "type": "text",
+            "validation": "text",
```
Comment (benmartin-coforma) [inline services/app-api/forms/sar.json:94]:
```suggestion
            "validation": "email",
```

File: services/app-api/forms/sar.json [hunk 2]
```diff
@@ -14,10 +14,143 @@
...
+            "props": {
+              "label": "Project director title"
+            }
+          },
+          {
+            "id": "generalInformation_projectDirectorEmail",
+            "type": "text",
+            "validation": "text",
```
Comment (benmartin-coforma) [inline services/app-api/forms/sar.json:139]:
```suggestion
            "validation": "email",
```

File: services/app-api/forms/sar.json
Comment (benmartin-coforma) [inline services/app-api/forms/sar.json:1]: COMMENT: This file contains two types of dashes, and they're not entirely consistent. For example, this object uses [an en dash](https://www.compart.com/en/unicode/U+2013) in the label, but a [hyphen](https://www.compart.com/en/unicode/U+002d) in the heading. This isn't a problem, necessarily, but it is a bit odd. If the en dash isn't a design requirement, I'd recommend hyphens throughout.
```json
    "id": "ret_mediumTermStayAges18to64",
    "type": "text",
    "validation": "text",
    "props": {
      "label": "Ages 18–64",
      "styleAsOptional": true,
      "heading": "Medium-term stay: 21-100 days"
    }
```

File: services/app-api/forms/sar.json [hunk 3]
```diff
@@ -118,24 +1291,217 @@
...
+                "type": "p",
+                "content": "For each initiative, recipients must also report on progress toward achieving the objective(s) identified in the MFP Work Plan. Progress towards these objectives indicates the state or territory's greater ability to provide HCBS instead of services in institutional settings. If your state or territory has not achieved the targets for performance measures or expected time frames for deliverables set in the MFP Work plan, use the following questions to explain the barriers or challenges that have hindered progress and describe plans to address them."
+              }
+            ]
+          }
+        },
+        "enterEntityDetailsButtonText": "Edit",
+        "dashboardTitle": "Report progress for each initiative",
+        "dashboardSubtitle": "Initiatives and Work Plan topics are auto-populated from your most recent approved Work Plan.",
+        "tableHeader": "Initaitve name <br/> Work Plan topic",
```
Comment (benmartin-coforma) [inline services/app-api/forms/sar.json:1495]:
```suggestion
        "tableHeader": "Initiative name <br/> Work Plan topic",
```

File: services/app-api/forms/sar.json [hunk 4]
```diff
@@ -145,10 +1511,153 @@
...
+                }
+              ]
+            }
+          },
+          {
+            "id": "oa_describeTechnicalAssitanceActivities",
+            "type": "textarea",
+            "validation": "text",
+            "props": {
+              "label": "Describe the technical assitance activities MFP staff have engaged in during the reporting period (e.g., participation in a learning collaborative or other training session)."
```
Comment (benmartin-coforma) [inline services/app-api/forms/sar.json:1624]:
```suggestion
              "label": "Describe the technical assistance activities MFP staff have engaged in during the reporting period (e.g., participation in a learning collaborative or other training session)."
```

**Review by benmartin-coforma — COMMENTED** [review]
I haven't tested this but I have read through the code and JSON, which overall looks very good!

File: services/ui-src/src/components/app/AppRoutes.tsx [hunk 5]
```diff
@@ -64,6 +64,7 @@ export const AppRoutes = () => {
               )}
             </>
           )}
+          )
```
Comment (gmrabian) [inline services/ui-src/src/components/app/AppRoutes.tsx:67]: remove this extra parenthesis

File: services/ui-src/src/components/fields/TextField.tsx [hunk 6]
```diff
@@ -84,6 +85,8 @@ export const TextField = ({
 
   return (
     <Box sx={sxOverride} className={`${nestedChildClasses} ${labelClass}`}>
+      {/* SAR field sections */}
+      {heading && <Heading sx={sx.fieldHeading}>{heading}</Heading>}
```
Comment (gmrabian) [inline services/ui-src/src/components/fields/TextField.tsx:89]: 🤔 is it possible to add this kind of thing at a higher level so it doesn't have to be added to each field type that requires it? might be over-engineering

File: services/app-api/forms/sar.json [hunk 7]
```diff
@@ -14,10 +14,143 @@
       "verbiage": {
         "intro": {
           "section": "",
-          "subsection": "",
-          "hint": "",
-          "exportSection": ""
+          "subsection": "General Information",
+          "hint": ""
```
Comment (gmrabian) [inline services/app-api/forms/sar.json:18]: I don't see `hint` in [types](https://github.com/Enterprise-CMCS/macpro-mdct-mfp/blob/main/services/ui-src/src/types/reports.ts#L16-L24) (I linked main, not your branch, but it's not in either)

It does render though 🤷

File: services/app-api/forms/sar.json [hunk 8]
```diff
@@ -30,9 +163,58 @@
...
+                "type": "text",
+                "validation": "text",
+                "props": {
+                  "label": "Number of I/DD"
+                }
+              },
+              {
+                "id": "ret_numberOfMHSUD",
+                "type": "text",
+                "validation": "text",
```
Comment (gmrabian) [inline services/app-api/forms/sar.json:212]: do we want these to be number validation types?

**Review by gmrabian — CHANGES_REQUESTED** [review]

File: services/app-api/forms/sar.json (hunk 8 above)
Comment (karla-vm) [reply services/app-api/forms/sar.json:212]: ooh, good point!

File: services/app-api/forms/sar.json
Comment (karla-vm) [reply services/app-api/forms/sar.json:1]: great find, I changed them to hyphens!

File: services/ui-src/src/components/fields/TextField.tsx (hunk 6 above)
Comment (karla-vm) [reply services/ui-src/src/components/fields/TextField.tsx:89]: yeah! I wasn't sure where to put this, so I just added it for now where it's being used, but I can cut a ticket to SPIKE it out 🤔

File: services/app-api/forms/sar.json [hunk 9]
```diff
@@ -14,10 +14,143 @@
...
+            "type": "text",
+            "validation": "text",
+            "props": {
+              "label": "MFP program's public name"
+            }
+          },
+          {
+            "id": "generalInformation_mfpProgramWebsite",
+            "type": "text",
+            "validation": "text",
```
Comment (gmrabian) [inline services/app-api/forms/sar.json:69]: maybe make this url validation type?

File: services/app-api/forms/sar.json [hunk 10]
```diff
@@ -107,9 +853,436 @@
...
+                  "choices": [
+                    {
+                      "id": "2VffASWS2XRfAlc3uLzxCVAC",
+                      "label": "Moved out of MFP jurisdiction/state/territory",
+                      "children": [
+                        {
+                          "id": "otherReasons_movedOutNumberOfOlderAdults",
+                          "type": "number",
+                          "validation": {
+                            "type": "number",
```
Comment (gmrabian) [inline services/app-api/forms/sar.json:991]: these nested fields are type: number but not erroring when I enter something invalid. likely not an issue with validation, but perhaps with when/how validation is triggering? might be out of scope for this PR

**Review by gmrabian — COMMENTED** [review]
Noticed that `Continue` button doesn't do anything if fields are filled out correctly, but it works when they are invalid or empty 😛 

anyway good stuff! let me know what you think about the comments and then I'm good to approve

File: services/app-api/forms/sar.json (hunk 10 above)
Comment (karla-vm) [reply services/app-api/forms/sar.json:991]: I don't think we've got validation set up just yet for this form! that might be the issue?

**Comment by karla-vm** [comment]
@gmrabian I think we're good to go on this one! 💯 great comments throughout, thanks for the thorough review. re: the `Continue` button, I think it's out of scope for this ticket, but we definitely need to hook up the routing correctly.

**Review by gmrabian — APPROVED** [review]
Great stuff!

**Pull request merged**